core/peerset/src/lib.rs

```diff
@@ -613,6 +623,33 @@ mod tests {
 		]);
 	}
 
+	#[test]
+	fn test_peerset_report_invulnerable() {
+		let bootnode = PeerId::random();
+		let bootnode2 = PeerId::random();
+		let reserved_node = PeerId::random();
+		let config = PeersetConfig {
+			in_peers: 0,
+			out_peers: 1,
+			bootnodes: vec![bootnode.clone(), bootnode2.clone()],
+			reserved_only: false,
+			reserved_nodes: vec![reserved_node.clone()],
+		};
+
+		let (mut peerset, handle) = Peerset::from_config(config);
+		handle.report_peer(bootnode2, -1);
+		handle.report_peer(reserved_node.clone(), -1);
+
+
+		while !peerset.message_queue.is_empty() {
+			let (message, p) = next_message(peerset).expect("Oh no");
+			assert_ne!(message, Message::Drop(reserved_node.clone()));
```

Comment:
Can we add something like below to ensure non-reserved nodes are actually dropped:
```rust
assert!(message, Message::Drop(bootnode2.clone()));
```

Reply:
Could follow other test cases:
```
let _peerset = assert_messages(peerset, vec![
    Message::Connect(reserved_node),
]);
```
This was passed.

Try to add `drop bootnode`
```
let _peerset = assert_messages(peerset, vec![
    Message::Connect(reserved_node),
    Message::Drop(bootnode),
]);
```
The test was new resolved

Reply:
Test case:
`test_peerset_report_peer()`
shows that non-reserved nodes are dropped